### ISSUE TITLE
EIP712 - encoding array types throws an error

### ISSUE THREAD

**Issue by gal432** [Mrtenz/eip-712#26]
Hi, 
it seems like the encoding for array fields doesn't work. 
Example for the typedData
```
{
  "types": {
    "EIP712Domain": [
      {
        "name": "name",
        "type": "string"
      },
      {
        "name": "version",
        "type": "string"
      },
      {
        "name": "chainId",
        "type": "uint256"
      },
      {
        "name": "verifyingContract",
        "type": "address"
      },
      {
        "name": "salt",
        "type": "bytes32"
      }
    ],
    "Person": [
      {
        "name": "name",
        "type": "string"
      },
      {
        "name": "wallet",
        "type": "address"
      }
    ],
    "Mail": [
      {
        "name": "from",
        "type": "Person[]"
      },
      {
        "name": "to",
        "type": "Person[]"
      }
    ]
  },
  "primaryType": "Mail",
  "domain": {
    "name": "Ether Mail",
    "version": "1",
    "chainId": 3,
    "verifyingContract": "0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC",
    "salt": "0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff"
  },
  "message": {
    "from": [
      {
        "name": "Alice",
        "wallet": "0xaaAAaaaAaaAAAAaaaaaAaaAAAaAAAAAAAaaAaaAA"
      }
    ],
    "to": [
      {
        "name": "Alice",
        "wallet": "0xaaAAaaaAaaAAAAaaaaaAaaAAAaAAAAAAAaaAaaAA"
      },
      {
        "name": "Bob",
        "wallet": "0xaaAAaaaAaaAAAAaaaaaAaaAAAaAAAAAAAaaAaaAA"
      }
    ]
  }
}
```

I used the following commands

```
const eip712 = require('eip-712');
eip712.getMessage(message, false);
```

And got the following error
```
Uncaught TypeError: Cannot read property 'map' of undefined
    at /app_dir/app/node_modules/eip-712/lib/cjs/eip-712.js:39:57
    at Array.map (<anonymous>)
    at encodeType (/app_dir/app/node_modules/eip-712/lib/cjs/eip-712.js:38:16)
    at getTypeHash (/app_dir/app/node_modules/eip-712/lib/cjs/eip-712.js:46:32)
    at encodeData (/app_dir/app/node_modules/eip-712/lib/cjs/eip-712.js:96:21)
    at getStructHash (/app_dir/app/node_modules/eip-712/lib/cjs/eip-712.js:103:32)
    at Object.getMessage (/app_dir/app/node_modules/eip-712/lib/cjs/eip-712.js:109:110)
```

**Comment by Mrtenz** [comment]
I'm able to reproduce. Looks like it breaks if you don't have a non-array type field. E.g., this works:

```json
{
  "Mail": [
      {
        "name": "from",
        "type": "Person"
      },
      {
        "name": "to",
        "type": "Person[]"
      },
      {
        "name": "contents",
        "type": "string[3]"
      }
    ]
}
```

But this doesn't:

```json
{
  "Mail": [
      {
        "name": "from",
        "type": "Person[]"
      },
      {
        "name": "to",
        "type": "Person[]"
      },
      {
        "name": "contents",
        "type": "string[3]"
      }
    ]
}
```

Note the type of `from` in both examples. I'll work on a solution for this problem.

**Comment by Mrtenz** [comment]
It turned out to be [a small typo](https://github.com/Mrtenz/eip-712/pull/27/files#diff-47d7fe55b92ddc9534e6f4b4553e2b40432a53dce561514200da9c22e9cc41f0L32-R32). Thanks for reporting this issue, the fixed version will be up on npm shortly.

**Comment by gal432** [comment]
Wow, you are really fast. thanks @Mrtenz !

**Comment by Mrtenz** [comment]
No problem! I just published `eip-712@0.4.3` on npm which should fix the issue.
